python/hail/docs/hailpedia/table.rst

```diff
@@ -0,0 +1,244 @@
+the dataset. Hail provides an :meth:`.Table.aggregate` method along with many
+aggregator functions (see :ref:`sec-aggregators`) to return the result of a
+query:
+
+    >>> ht.aggregate(agg.fraction(ht.SEX == 'F'))
+    0.5
+
+We also might want to compute the mean value of `HT` for each sex. This is
+possible with a combination of :meth:`Table.group_by` and
+:meth:`.GroupedTable.aggregate`:
+
+    >>> ht_agg = (ht.group_by(ht.SEX)
+    ...             .aggregate(mean = agg.mean(ht.HT)))
+    >>> ht_agg.show()
+    +-----+-------------+
+    | SEX |        mean |
+    +-----+-------------+
+    | str |     float64 |
+    +-----+-------------+
+    | M   | 6.85000e+01 |
+    | F   | 6.50000e+01 |
+    +-----+-------------+
+
+
+Note that the result of ``ht.group_by(...).aggregate(...)`` is a new
+:class:`.Table` while the result of ``ht.aggregate(...)`` is a Python value.
+
+Joins
+=====
+
+To join the row fields of two tables together, Hail provides a
+:meth:`.Table.join` method with options for how to join the rows together (left,
+right, inner, outer). The tables are joined by the row fields designated as
+keys. The number of keys and their types must be identical between the two
+tables. However, the names of the keys do not need to be identical. Use the
+:attr:`.Table.key` attribute to view the current table row keys and the
+:meth:`.Table.key_by` method to change the table keys. If top level row field
+names overlap between the two tables, the second table's field names will be
+appended with a unique identifier "_N".
+
+    >>> ht = ht.key_by('ID')
+    >>> ht2 = hl.import_table("data/kt_example2.tsv", impute=True).key_by('ID')
+
+    >>> ht_join = ht.join(ht2)
+    >>> ht_join.show()
+    +-------+-------+-----+-------+-------+-------+-------+-------+-------+--------+
+    |    ID |    HT | SEX |     X |     Z |    C1 |    C2 |    C3 |     A | B      |
+    +-------+-------+-----+-------+-------+-------+-------+-------+-------+--------+
+    | int32 | int32 | str | int32 | int32 | int32 | int32 | int32 | int32 | str    |
+    +-------+-------+-----+-------+-------+-------+-------+-------+-------+--------+
+    |     3 |    70 | F   |     7 |     3 |    10 |    81 |    -5 |    70 | mouse  |
+    |     4 |    60 | F   |     8 |     2 |    11 |    90 |   -10 |    60 | rabbit |
+    |     2 |    72 | M   |     6 |     3 |     2 |    61 |     1 |    72 | dog    |
+    |     1 |    65 | M   |     5 |     4 |     2 |    50 |     5 |    65 | cat    |
+    +-------+-------+-----+-------+-------+-------+-------+-------+-------+--------+
+
+In addition to using the :meth:`.Table.join` method, Hail provides an additional
+join syntax using Python's bracket notation. This syntax does a left join, like
+looking up values in a dictionary. Instead of returning a :class:`.Table`, this
+syntax returns an :class:`.Expression` which can be used in expressions of the
+left table. For example, below we add the field 'B' from `ht2` to `ht`:
```

Comment:
"of the left table" is confusing because `ht` appears syntactically to the right of `ht2`. Perhaps say "returns an :class:`.Expression` which can be used to annotate the indexing table with data from the indexed table"